lib/tpcal.py

```diff
@@ -1,33 +1,34 @@
+
 import lvgl as lv
 lv.init()
 
-lv.log_register_print_cb(lambda level,path,line,msg: print('LOG: %s(%d): %s' % (path, line, msg)))
-
 # Initialize ILI9341 display
 
 import lvesp32
-from ili9XXX import ili9341
-disp = ili9341()
-
-HRES = lv.disp_get_hor_res(lv.disp_t.cast(None))
-VRES = lv.disp_get_ver_res(lv.disp_t.cast(None))
-
-# Register raw resistive touch driver
-'''
-import rtch
-touch = rtch.touch(xp = 32, yp = 33, xm = 25, ym = 26, touch_rail = 27, touch_sense = 33, cal_x0=0, cal_x1 = HRES, cal_y0=0, cal_y1 = VRES)
-touch.init()
-indev_drv = lv.indev_drv_t()
-lv.indev_drv_init(indev_drv) 
-indev_drv.type = lv.INDEV_TYPE.POINTER;
-indev_drv.read_cb = touch.read;
-lv.indev_drv_register(indev_drv);
-'''
+import espidf as esp
+from ili9XXX import ili9341,COLOR_MODE_BGR,LANDSCAPE,PORTRAIT
+
+# select the orientation
+# orientation = LANDSCAPE
+orientation = PORTRAIT
+w=320
+h=240
+if orientation == PORTRAIT:
+    w,h = h,w
+    
+disp = ili9341(miso=19,mosi=23,clk=18, cs=26, dc=5, rst=-1, power=-1, backlight=-1, backlight_on=0, power_on=0,
+               spihost=esp.VSPI_HOST, mhz=40, factor=4, hybrid=True, width=w, height=h,
+               colormode=COLOR_MODE_BGR, rot=orientation, invert=False, double_buffer=True, half_duplex=True)
 
+HRES = disp.width
+VRES = disp.height
 # Register xpt touch driver
-import xpt2046 as xpt2046
-touch = xpt2046.xpt2046(cal_x0=0, cal_x1 = HRES, cal_y0=0, cal_y1 = VRES)
-
+from xpt2046 import xpt2046
+if orientation == LANDSCAPE:
+    touch = xpt2046(spihost=esp.VSPI_HOST,cal_x0=0, cal_x1 = HRES, cal_y0=0, cal_y1 = VRES, transpose=False)
```

Comment:
Why changed parameters?  
Why duplicate the entire xpt2046 construction, instead of passing a different "transpose" parameter to a single xpt2046 construction?

Reply:
I changed the parameters again to adapt to my hardware.
I now create a transp variable which I set up depending on landscape or portrait and pass this to the xpt2046, as you proposed

Reply:
> I changed the parameters again to adapt to my hardware.

I suggest not changing them in the script, unless there is a good reason to.